common-pg/src/main/scala/com/socrata/pg/analyzer2/rollup/SoQLSplitAnd.scala

```diff
@@ -0,0 +1,28 @@
+package com.socrata.pg.analyzer2.rollup
+
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.collection.NonEmptySeq
+import com.socrata.soql.functions.SoQLFunctions
+import com.socrata.soql.types.{SoQLType, SoQLValue}
+
+import com.socrata.pg.analyzer2.SqlizerUniverse
+
+class SoQLSplitAnd[MT <: MetaTypes with ({type ColumnType = SoQLType; type ColumnValue = SoQLValue})]
+    extends SplitAnd[MT]
+{
+  private val And = SoQLFunctions.And.monomorphic.get
+
+  override def split(e: Expr): NonEmptySeq[Expr] =
+    e match {
+      case FunctionCall(And, args) =>
+        val nonEmptyArgs = NonEmptySeq.fromSeq(args).getOrElse {
+          throw new Exception("AND with no arguments??")
```

Comment:
Hrm. Generic Exception type, but I guess it's really hard for this to happen....

Reply:
It should be impossible to get past analysis while passing no parameters to AND